File: Data Structures & Algorithms/find-duplicate-integer/submission-3.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        // int slow = nums[0];
        // int fast = nums[nums[0]];
        // while(slow != fast) {
        //     slow = nums[slow];
        //     fast = nums[nums[fast]];
        // }

        // slow = nums[slow];
        // int slow2 = nums[0];
        // while(slow != slow2) {
        //     slow = nums[slow];
        //     slow2 = nums[slow2];
        // }
        // return slow;

        int curr = 0;
        while(nums[curr] > 0) {
            nums[curr] = -1*nums[curr];
            curr = abs(nums[curr]);
        }
        return curr;
    }
};

File: Data Structures & Algorithms/majority-element/submission-3.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        d = {}
        n = (len(nums) // 2) + 1
        for i in nums:
            if i not in d:
                d[i] = 0
            d[i] += 1
            if d[i] == n:
                return i
        return -1
            